### PR TITLE
Feature/change infinity scroll to cdk virtual scroll on select search and select multi tags

### PR DESCRIPTION
**Pull request by JoaoBianco**
https://duopana.myjetbrains.com/youtrack/issue/SQ-66102/FRONT-Avaliar-Remocao-de-Infinity-Scroll-e-implementacao-do-cdkVirtualScroll-no-sq-select-multi-tags
https://duopana.myjetbrains.com/youtrack/issue/SQ-66101/FRONT-Avaliar-Remocao-de-Infinity-Scroll-e-implementacao-do-cdkVirtualScroll-no-sq-select-search

## Release Notes

- **New Features**
  - Introduced virtual scrolling for improved performance in select components.
  - Added a new CSS class for better styling of option lists.

- **Bug Fixes**
  - Simplified option management by removing unnecessary pagination logic.

- **Documentation**
  - Updated the public API to reflect changes in the component's properties and methods.

- **Chores**
  - Restructured module imports to streamline component management.

### CODE REVIEW

File: src/components/sq-select-multi-tags/sq-select-multi-tags.component.html [hunk 1]
```diff
@@ -90,19 +90,11 @@
...
-          (scrolledEmitter)="addMoreOptions()"
-        >
-          <ng-container *ngFor="let opt of _options | searchFromAlternativeArray:searchText:options; let i = index; trackBy: trackByOptValue">
-            <ng-template *ngTemplateOutlet="option; context: { opt: opt, i: i }"></ng-template>
+        <cdk-virtual-scroll-viewport itemSize="32" class="list">
+          <ng-container *cdkVirtualFor="let opt of _options | search:searchText; i as index; trackBy: trackByOptValue">
+            <ng-template *ngTemplateOutlet="option; context: { opt: opt, i: index }"></ng-template>
           </ng-container>
-        </sq-infinity-scroll>
+        </cdk-virtual-scroll-viewport>
```
[the remark this answers is not in the document]
Comment (JoaoBianco) [reply src/components/sq-select-multi-tags/sq-select-multi-tags.component.html:97]: In this case, we can use 32px

File: src/components/sq-select-multi-tags/sq-select-multi-tags.component.scss [hunk 2]
```diff
@@ -138,6 +138,10 @@
     position: relative;
   }
 
+  .list {
+    height: 305px;
```
Comment (JonasPeres) [inline src/components/sq-select-multi-tags/sq-select-multi-tags.component.scss:142]: A lista já não tem um tamanho fixo?

File: src/components/sq-select-search/sq-select-search.component.scss [hunk 3]
```diff
@@ -97,6 +97,7 @@
     margin: 0;
     width: 100%;
     margin: 0.7rem 0 0;
+    height: 305px;
```
Comment (JonasPeres) [inline src/components/sq-select-search/sq-select-search.component.scss:100]: Achei que já tinha a altura fixa nos dois

**Review by JonasPeres — CHANGES_REQUESTED** [review]
Precisa passar em cada um deles no portal e ver como vai ficar

**Review by JonasPeres — DISMISSED** [review]

**Review by JonasPeres — DISMISSED** [review]

**Review by JonasPeres — APPROVED** [review]

**Pull request merged**